### PR TITLE
feat: added kubernetes replicator addon and test

### LINKED ISSUE #201: Add https://github.com/mittwald/kubernetes-replicator to the initium-platform
We need a system to replicate secrets used by the application on each and every namespace generate by initium.

This story is just about adding the component to the platform and potentially writing a test for it.

### PR DESCRIPTION
**Pull request by jeevanions**
## What does this PR do?

- New addon added `kubernetes-replicator` added to initium platform.
- Test added to test the replicator 

## Related issues

closes #201

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -87,6 +87,18 @@ if os.getenv('INITIUM_DEPLOY_MINIMAL', 'false') == 'false':
...
+    ## Kubernetes replicator
+    k8s_yaml(namespace_yaml('kubernetes-replicator'), allow_duplicates=False)
+    local('helm dependency update ./addons/kubernetes-replicator')
+    local('helm repo update')
+
+    k8s_yaml(helm(
+        './addons/kubernetes-replicator',
+        name='kubernetes-replicator',
+        namespace='kubernetes-replicator',
+    ))
```
Comment (LucaLanziani) [inline Tiltfile:99]: @jeevanions this should not be here, the replicator is installed as part of the addons using the app of apps

Comment (LucaLanziani) [reply Tiltfile:99]: the addon is missing from the `app-of-apps/values.yaml` file, you should add it there the get it installed

File: tests/kubernetes_replicator_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+// 	helmOptions, err := prepareHelmEnvironment(t, &addonData)
+
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+
+// 	destroyHelmEnvironment(t, addonData, helmOptions)
+// 	defer k8s.DeleteNamespace(t, kubectlOptions, "kubernetes-replicator")
+
+// }
```
Comment (LucaLanziani) [inline tests/kubernetes_replicator_test.go:45]: No need to keep this around if not used 
```suggestion
```

Comment (jeevanearform) [reply tests/kubernetes_replicator_test.go:45]: Addressed

File: Tiltfile [hunk 3]
```diff
@@ -87,6 +87,9 @@ if os.getenv('INITIUM_DEPLOY_MINIMAL', 'false') == 'false':
         namespace='argocd',
     ))
 
+
```
Comment (zaremba-tomasz) [inline Tiltfile:90]: Left over, right?

**Review by LucaLanziani — APPROVED** [review]

**Pull request merged**